### PR TITLE
conan: 1.12.0 -> 1.12.3

### LINKED ISSUE #68587: Can we deprecate pyyaml_3?
In https://github.com/NixOS/nixpkgs/pull/68481 the pinned pyyaml_3 was removed since the package works now with the current version.

It is still used in other packages. Check if it is still needed.

https://github.com/NixOS/nixpkgs/search?q=pyyaml_3&unscoped_q=pyyaml_3

When no package needs it anymore, wen can deprecate pyyaml_3.

cc maintainers @HaoZeke @muflax

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
partially resolve #68587

I also added tests, the checkPhase just didn't do anything before

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

```
https://github.com/NixOS/nixpkgs/pull/73696
1 package were build:
conan
```

### CODE REVIEW

File: pkgs/development/tools/build-managers/conan/default.nix [hunk 1]
```diff
@@ -37,43 +37,48 @@ let newPython = python3.override {
...
-  propagatedBuildInputs = with newPython.pkgs; [
-    colorama deprecation distro fasteners bottle
-    future node-semver patch pygments pluginbase
-    pyjwt pylint pyyaml requests six tqdm
-  ];
-
   checkPhase = ''
-    export HOME="$TMP/conan-home"
-    mkdir -p "$HOME"
+    export HOME=$TMPDIR
```
Comment (HaoZeke) [inline pkgs/development/tools/build-managers/conan/default.nix]: $TMPDIR is not set on most distributions

**Review by HaoZeke — CHANGES_REQUESTED** [review]
Everything else looks fine, but I would prefer a path set from $HOME for the build. Thanks for this!!

File: pkgs/development/tools/build-managers/conan/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/development/tools/build-managers/conan/default.nix]: This is the check phase, it is set for nix builds

Comment (HaoZeke) [reply pkgs/development/tools/build-managers/conan/default.nix]: Oh, sorry, yup, just checked. Thanks!

**Review by HaoZeke — APPROVED** [review]

File: pkgs/development/tools/build-managers/conan/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/development/tools/build-managers/conan/default.nix]: No problem :)

**Pull request merged**